### PR TITLE
add syndicate recruiter midround event

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
adds a recruiter event, their job is to interview people on the station and hire the best candidates for the syndicate. ask some questions and write the answers, do up a contract whatever you want works
all that matters is that once you have the paperwork done, the candidate uses your special recruiter pen to prick their finger and sign it with their blood. this gives you the sweet greentext but since its an mrp server you are expected to actually do some recruiting and not nrp postal 2 "will you please sign my petition".

you also arent allowed to recruit mindshielded people or yourself for obvious reasons. if you want to recruit an ex-seccie get an implanter to remove the mindshield :trollface:

the objectives are simple:
- escape
- recruit a random number of people using the mechanism above
- keep your pen safe, it has the dna of everyone you recruited

## Why / Balance
had a lot of good rp with it as selfantag, now its realantag :trollface:

the RecruiterPen component can also be reused for non syndicate (e.g. tgmc recruiter) admeme events with uploaded prototypes

## Technical details
- added `SignAttemptEvent` for signatures to get cancelled, used by recruiter pen
- `RecruiterPen` component makes a pen need blood to sign paper and lets you prick your finger to refill it with blood

## Media
rules should make it reasonably clear what you have to and cant do

valid...

the recruiter ship

signing with blood is real

only recruited 1 person enjoy the redtext bozo

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
no

**Changelog**
:cl:
- add: There are now reports of a syndicate recruitment drive in NT sectors.

### CODE REVIEW

**Comment by dvir001** [comment]
Time to hire punpun

**Comment by noctyrnal** [comment]
I feel like adding the recruitment objective to regular agents and letting them buy 1 pen for free could be a cool RP-friendly objective for them too

**Comment by WarMechanic** [comment]
image

will the syndicate recruiter have a way to give new employees syndicate clothes?

**Comment by deltanedas** [comment]
> will the syndicate recruiter have a way to give new employees syndicate clothes?

no its just "you will get mission and gear after centcom" meme right now, but the ship could be expanded with some valid gear
though it would make it harder to set up shop somewhere on station because then either you have to go back to the ship for the drip or bring it with you and be giga sus

File: Content.Shared/DeltaV/Recruiter/SharedRecruiterPenSystem.cs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        // all checks passed, let it be signed!
+        args.Cancelled = false;
+    }
+
+    private bool CheckBlacklist(Entity<RecruiterPenComponent> ent, EntityUid user, string action)
+    {
+        if (!Mind.TryGetMind(user, out var mind, out _))
+            return false; // mindless nt drone...
+
+        // TODO: when upstream merged make this use whitelist system blacklist helper functions
```
Comment (Axionyxx) [inline Content.Shared/DeltaV/Recruiter/SharedRecruiterPenSystem.cs]: TODO pro

**Review by Axionyxx — CHANGES_REQUESTED** [review]
Upstream is merged

**Review by Axionyxx — DISMISSED** [review]
Fix the conflict and I'll merge it

**Review by Axionyxx — APPROVED** [review]

**Pull request merged**